### DISCUSSION TITLE
AWS Lambda - Container

### DISCUSSION THREAD

**Discussion by stephenbawks** [Kludex/mangum#166] [Q&A] [answered]
Trying out building a quick and simple app to use with the HTTP API and AWS Lambda with a container.  I can set up the app locally and hit it locally, however I am not sure if I am missing something but I cannot get it working in Lambda with a container.

Super simple main.py file that looks like the following:

```
import json
import boto3
import os
import uvicorn

from mangum import Mangum
from fastapi import FastAPI
from pydantic import BaseModel


app = FastAPI()

@app.get("/")
def read_root():
    return {"Hello": "World"}

@app.get("/items/{item_id}")
def read_item(item_id: int, q: str = None):
    return {"item_id": item_id, "q": q}


handler = Mangum(app)

```

With my Dockerfile it looks like the following:

```
# Define function directory
FROM public.ecr.aws/lambda/python:3.8

RUN pip install poetry

RUN mkdir /app
COPY ./app /app

COPY ./poetry.lock  ./pyproject.toml ./
RUN poetry install --no-dev

CMD [ "app.main.handler" ]
```

I created a HTTP API and basically just have a single proxy route that points to my Lambda.
```
ANY /{proxy+}
```

I am not sure if the handler part is correct for the container or does it need to be a uvicorn command to run the app.main:app file?  The error I was seeing in Cloudwatch looks like this.

```
[ERROR] Runtime.ImportModuleError: Unable to import module 'app.main': No module named 'app'
```

**Comment by koxudaxi** [comment, marked as answer]
@stephenbawks 
Did you create `__init__.py` in the `app` directory?
```
- /poetry.lock
   pyproject.toml
   Dockerfile
   app/__init__.py
       main.py
```
And,  I fixed your Dockerfile.
```Dockerfile
# Define function directory
FROM public.ecr.aws/lambda/python:3.8

RUN pip install poetry

COPY ./app /var/task/app

COPY ./poetry.lock  ./pyproject.toml ./
RUN poetry config virtualenvs.create false && poetry install --no-dev

CMD [ "app.main.handler" ]

```
You should think about how to use poetry. this example may not best practice on poetry.
Thank you.

**Comment by jordaneremieff** [comment]
Thanks @koxudaxi.

@stephenbawks closing this because it doesn't seem related to Mangum itself, but feel free to continue discussion if the previous comment did not clear thing things up.
